src/charm.py

```diff
@@ -150,9 +154,11 @@ def server_cert_path(self) -> Optional[str]:
     def internal_url(self) -> str:
         """Returns workload's FQDN. Used for ingress."""
         scheme = "http"
+        port = "8080"
         if hasattr(self, "coordinator") and self.coordinator.nginx.are_certificates_on_disk:
             scheme = "https"
-        return f"{scheme}://{self.hostname}:8080"
+            port = "443"
```

Comment:
- In prometheus and others, we keep using the same port (e.g. 9090) even when TLS is enabled.
Is changing ports necessary? Should we keep using the same port even when TLS is enabled, like we do for prom, loki, grafana?
- Iirc, there's a pattern where apps use 8080/8443 instead of 80/443. Doesn't really matter, just the 8080 and 443 combo seems odd.
- If we use 80/443, then a scheme would be enough - we won't need to specify ports.

Reply:
I don't mind either, I would discuss this briefly in a team café to make sure we're all on the same page, as this would involve changes in Tempo as well